qos-monitor/src/main/java/eu/arrowhead/core/qos/service/QoSMonitorDriver.java

```diff
@@ -508,24 +510,67 @@ private SystemRequestDTO getQosMonitorSystemRequestDTO() {
 
 		final PublicKey publicKey = (PublicKey) arrowheadContext.get(CommonConstants.SERVER_PUBLIC_KEY);
```

Comment:
You should move this line inside the if block. Or even better if you move inside the sslProperties.isSslEnabled() if block.

Reply:
done